rust/sealevel/programs/ism/multisig-ism-message-id/src/metadata.rs

```diff
@@ -32,7 +34,12 @@ impl TryFrom<Vec<u8>> for MultisigIsmMessageIdMetadata {
         }
 
         let origin_mailbox = H256::from_slice(&bytes[ORIGIN_MAILBOX_OFFSET..MERKLE_ROOT_OFFSET]);
-        let merkle_root = H256::from_slice(&bytes[MERKLE_ROOT_OFFSET..SIGNATURES_OFFSET]);
+        let merkle_root = H256::from_slice(&bytes[MERKLE_ROOT_OFFSET..MERKLE_INDEX_OFFSET]);
+        // This cannot panic since SIGNATURES_OFFSET - MERKLE_INDEX_OFFSET is 4.
+        let merkle_index_bytes: [u8; 4] = bytes[MERKLE_INDEX_OFFSET..SIGNATURES_OFFSET]
+            .try_into()
+            .unwrap();
```

Comment:
I don't feel strongly but noting that we return a Result here anyways so we could maybe do that instead of unwrapping? will defer to you though